misc-utils/lsfd-unkn.c

```diff
@@ -1223,6 +1274,14 @@ static char *anon_bpf_map_get_name(struct unkn *unkn)
 		xasprintf(&str, "id=%d type=%s", data->id, t);
 	else
 		xasprintf(&str, "id=%d type=UNKNOWN(%d)", data->id, data->type);
+
+	if (*data->name) {
+		char name[6	/* " name=" */
+			  + BPF_OBJ_NAME_LEN + 1];
+		snprintf(name, sizeof(name), " name=%s", data->name);
+		xstrappend(&str, name);
```

Comment:
For my own code I wondered why there is no `xstrappend_realloc`.
This would also have helped here.

Reply:
If I understand your comment correctly, I think what we need (in this context) is xasprintf_realloc or xasprintf_append. With the function, my change can be simplified like:
```diff
diff --git a/misc-utils/lsfd-unkn.c b/misc-utils/lsfd-unkn.c
index 9c2ffdd09..55f6851e0 100644
--- a/misc-utils/lsfd-unkn.c
+++ b/misc-utils/lsfd-unkn.c
@@ -1274,12 +1274,8 @@ static char *anon_bpf_map_get_name(struct unkn *unkn)
 	else
 		xasprintf(&str, "id=%d type=UNKNOWN(%d)", data->id, data->type);
 
-	if (*data->name) {
-		char name[6	/* " name=" */
-			  + BPF_OBJ_NAME_LEN + 1];
-		snprintf(name, sizeof(name), " name=%s", data->name);
-		xstrappend(&str, name);
-	}
+	if (*data->name)
+		xasprintf_realloc(&str, " name=%s", data->name);
 
 	return str;
 }
```

Am I correct?

Reply:
(Work in progress)

Reply:
I think what we need is
```
int strfappend(const char **a, const char *format, ...)    
```

. a hybrid version of strappend and strfconcat.

Reply:
To lib/strutils, I added strfappend and strvfappend.
To lsfd.h, I added xstrfappend and xstrvfappend.

I didn't add xstrfappend and xstrvfappend to xalloc.h because the header file doesn't include strutils.h.